[ANE-283] Adds sbt dependency tree strategy (#942)
fossas/fossa-cli: docs/references/strategies/languages/scala/sbt.md

While the other analysis strategies for `gradle` and `maven` offer some scala project coverage, scala projects overwhelmingly use the build tool `sbt`.

| Strategy           | Direct Deps        | Deep Deps          | Edges              | Other Limitations                                                                                       |
| ------------------ | ------------------ | ------------------ | ------------------ | ------------------------------------------------------------------------------------------------------- |

> | sbt dependencyTree | :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark: | Requires sbt v1.4 or greater. Versions do not include classifiers. Does not support multi-project build |

Reviewer: Should we mention that you can also install the plugin to make this work?

## Analysis

1. Run `sbt makePom` to generate pom files

## Strategy: `sbt dependencyTree`

> 1. From generated pom file, identify project name via, `<name>` attribute.
> 2. Perform `sbt $project/dependencyTree` and parse it's output to create dependency graph
> 
> With [sbt 1.4.0 release](https://www.scala-sbt.org/1.x/docs/sbt-1.4-Release-Notes.html#sbt-dependency-graph+is+in-sourced), `dependencyTree` command is available by default. If you are using
> older sbt version, you can install following plugin: https://github.com/sbt/sbt-dependency-graph, which will also enable fossa-cli to work with `dependencyTree` task.

Reviewer: ```suggestion
1. From the generated pom file, identify the project name via the `<name>` attribute.
2. Perform `sbt $project/dependencyTree` and parse its output to create a dependency graph

With [sbt 1.4.0 release](https://www.scala-sbt.org/1.x/docs/sbt-1.4-Release-Notes.html#sbt-dependency-graph+is+in-sourced), the `dependencyTree` command is available by default. If you are using
an older sbt version, you can install the following plugin: https://github.com/sbt/sbt-dependency-graph. This will also enable fossa-cli to work with the `dependencyTree` task.
```

For sbt < 1.3:
```
# in project/plugins.sbt
addSbtPlugin("net.virtual-void" % "sbt-dependency-graph" % "0.10.0-RC1")
```

> > Unfortunately, `sbt dependencyTree` command does not include version classifier in it's output. Further, 
> > currently we do not support this strategy for multi-project builds due [inconsistent reporting defect in sbt](https://github.com/sbt/sbt/issues/6905)

Reviewer: ```suggestion
> Unfortunately, the `sbt dependencyTree` command does not include version classifier in its output. Further, 
> currently we do not support this strategy for multi-project builds due an [inconsistent reporting defect in sbt](https://github.com/sbt/sbt/issues/6905).
```
